### PR TITLE
Query osmo equivilent is staked via superfluid

### PR DESCRIPTION
**Pull request by hieuvubk**
Closes: #1525

## What is the purpose of the change

>  Cli command
> Rpc Query: _SuperfluidOSMODelegationsByDelegator_ same as _SuperfluidDelegationsByDelegator_  but return osmo value
> Add test

### CODE REVIEW

File: proto/osmosis/superfluid/query.proto [hunk 1]
```diff
@@ -70,6 +70,15 @@ service Query {
                                    "superfluid_delegations/{delegator_address}";
   }
 
+  // Returns all the superfluid poistions for a specific delegator
+  rpc SuperfluidOSMODelegationsByDelegator(
```
Comment (alexanderbez) [inline proto/osmosis/superfluid/query.proto:74]: This is a mouthful haha. Any chance we can think of more abbreviated nomenclature.

Comment (hieuvubk) [reply proto/osmosis/superfluid/query.proto:74]: he we can think of better names. How do u think about the logic

Comment (hieuvubk) [reply proto/osmosis/superfluid/query.proto:74]: @mattverse want to create  a separate query (as i understant). Refer to pr https://github.com/osmosis-labs/osmosis/pull/1539

Comment (hieuvubk) [reply proto/osmosis/superfluid/query.proto:74]: Should change to _SuperfluidEquivilentDelegationsByDelegator_?

File: proto/osmosis/superfluid/query.proto [hunk 2]
```diff
@@ -187,6 +187,10 @@ message SuperfluidDelegationsByDelegatorResponse {
     (gogoproto.nullable) = false,
     (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
   ];
+  cosmos.base.v1beta1.Coin total_equivilent_staked_amount = 3 [
```
Comment (alexanderbez) [inline proto/osmosis/superfluid/query.proto:190]:
```suggestion
  cosmos.base.v1beta1.Coin total_equivalent_staked_amount = 3 [
```

**Review by alexanderbez — COMMENTED** [review]
Looks much better :P

File: x/superfluid/keeper/grpc_query.go [hunk 3]
```diff
@@ -213,17 +216,25 @@ func (q Querier) SuperfluidDelegationsByDelegator(goCtx context.Context, req *ty
 		baseDenom := periodLock.Coins.GetDenomByIndex(0)
 		lockedCoins := sdk.NewCoin(baseDenom, periodLock.GetCoins().AmountOf(baseDenom))
 		valAddr, err := ValidatorAddressFromSyntheticDenom(syntheticLock.SynthDenom)
+
+		// Find how many osmo tokens this delegation is worth at superfluids current risk adjustment
+		// and twap of the denom.
+		equivilentAmount := q.Keeper.GetSuperfluidOSMOTokens(ctx, baseDenom, lockedCoins.Amount)
+		coin := sdk.NewCoin(appparams.BaseCoinUnit, equivilentAmount)
```
Comment (ValarDragon) [inline x/superfluid/keeper/grpc_query.go:223]:
```suggestion
		equivalentAmount := q.Keeper.GetSuperfluidOSMOTokens(ctx, baseDenom, lockedCoins.Amount)
		coin := sdk.NewCoin(appparams.BaseCoinUnit, equivalentAmount)
```

minor nit

File: docs/core/proto-docs.md [hunk 4]
```diff
@@ -3691,6 +3692,7 @@ assets
 | ----- | ---- | ----- | ----------- |
 | `superfluid_delegation_records` | [SuperfluidDelegationRecord](#osmosis.superfluid.SuperfluidDelegationRecord) | repeated |  |
 | `total_delegated_coins` | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) | repeated |  |
+| `total_equivilent_staked_amount` | [cosmos.base.v1beta1.Coin](#cosmos.base.v1beta1.Coin) |  |  |
```
Comment (ValarDragon) [inline docs/core/proto-docs.md:3695]: I think this needs to be regenerated due to the typo

**Review by ValarDragon — APPROVED** [review]
LGTM once proto-doc / typo is sorted out.

Thanks for adding this feature! Want to make a Changelog update? (Feature in unreleased)

**Comment by hieuvubk** [comment]
> LGTM once proto-doc / typo is sorted out.
> 
> Thanks for adding this feature! Want to make a Changelog update? (Feature in unreleased)

Thank u, i missed. Also add entry to changelog.
Btw i dont know why running make proto-all, it's not rebuild the proto-docs 🤔

**Pull request merged**